### PR TITLE
Add support for selectors and overrides in Azure Policy CRUD cmdlets

### PR DESCRIPTION
**Pull request by mentat9**
## Description

Implement ResourceSelector and Override parameters for New/Update-AzPolicyAssignment
Implement ResourceSelector parameter for New/Update-AzPolicyExemption
Add new tests for selectors and overrides
Add new examples for selectors and overrides

## Mandatory Checklist

- Please choose the target release of Azure PowerShell. (⚠️**Target release** is a different concept from **API readiness**. Please click below links for details.)
  - [X] [General release](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Public preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Private preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Engineering build](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] No need for a release

- [X] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For SDK-based development mode, update `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`.
        * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. 
    * For autorest-based development mode, include the changelog in the PR description.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/Resources/Policy.Autorest/docs/New-AzPolicyAssignment.md [hunk 1]
```diff
@@ -185,7 +213,7 @@ The commands store that object in the $AllowedLocations variable.
 The final command assigns the policy in $Policy at the level of a resource group using the policy parameter object in $AllowedLocations.
 The **ResourceId** property of $ResourceGroup identifies the resource group.
 
-### Example 10: [Backcompat] Policy assignment at resource group level with policy parameter file
+### Example 12: [Backcompat] Policy assignment at resource group level with policy parameter file
 ```powershell
 {
```
Comment (msJinLei) [inline src/Resources/Policy.Autorest/docs/New-AzPolicyAssignment.md:218]: Powershell section can only includes powershell scripts

**Review by msJinLei — CHANGES_REQUESTED** [review]

File: src/Resources/Policy.Autorest/docs/New-AzPolicyAssignment.md (hunk 1 above)
Comment (mentat9) [reply src/Resources/Policy.Autorest/docs/New-AzPolicyAssignment.md:218]: > Powershell section can only includes powershell scripts

Updated examples.

**Comment by mentat9** [comment]
@msJinLei - I've applied your feedback and removed the needs-revision label. Can you take another look? The CI check failures seem to be from checkers crashing: not sure how to debug that.

**Comment by msJinLei** [comment]
> @msJinLei - I've applied your feedback and removed the needs-revision label. Can you take another look? The CI check failures seem to be from checkers crashing: not sure how to debug that.

You the files of `doc` folder is generated. Please fix what of `examples` folder and regenerate the module

**Comment by mentat9** [comment]
@msJinLei - My last change should fix the test failures. Please take another look when you get a chance.

**Review by msJinLei — APPROVED** [review]

**Pull request merged**